### PR TITLE
Expose The Stress-Ng CommandLine Parameters to Profile

### PR DESCRIPTION
**Pull request by imadityaa**

### CODE REVIEW

File: src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs [hunk 1]
```diff
@@ -158,10 +181,27 @@ private async Task CaptureMetricsAsync(IProcessProxy process, string commandArgu
 
         private string GetCommandLineArguments()
         {
-            int coreCount = Environment.ProcessorCount;
+            string commandLineArgs = this.CommandLine;
+
+            if (!commandLineArgs.Contains("--cpu"))
```
Comment (nmalkapuram) [inline src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs:186]: Even in the scenario where we need to use only some % of CPU, is this parameter present in the command line?

File: src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs [hunk 2]
```diff
@@ -49,13 +49,13 @@ public StressNgExecutor(IServiceCollection dependencies, IDictionary<string, ICo
...
         /// </summary>
-        public int DurationInSecond
+        public string CommandLine
         {
             get
             {
-                 return this.Parameters.GetValue<int>(nameof(StressNgExecutor.DurationInSecond), DefaultRuntimeInSeconds);
+                return this.Parameters.GetValue<string>(nameof(StressNgExecutor.CommandLine));
             }
         }
```
Comment (nmalkapuram) [inline src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs:61]: Update the summary for the IntializeAsync() method below.

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-STRESSNG.json [hunk 3]
```diff
@@ -4,15 +4,18 @@
     "Metadata": {
         "RecommendedMinimumExecutionTime": "00:05:00",
         "SupportedPlatforms": "linux-x64,linux-arm64",
-        "SupportedOperatingSystems": "CBL-Mariner,CentOS,Debian,RedHat,Suse,Ubuntu"
+        "SupportedOperatingSystems": "CBL-Mariner,CentOS,Debian,RedHat,Suse,Ubuntu",
+        "SpecialInstrucions":  "The commandline parameter --yaml is programatically added, do not add it in profile. By default --cpu is set to ProcessorCount."
```
Comment (deep1712) [inline src/VirtualClient/VirtualClient.Main/profiles/PERF-STRESSNG.json:8]: We can add information about default timeout , cpu threads and metrics flag is on.

Comment (imadityaa) [reply src/VirtualClient/VirtualClient.Main/profiles/PERF-STRESSNG.json]: Modified the Special Instructions

File: src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs (hunk 2 above)
Comment (imadityaa) [reply src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs:61]: Corrected the summary

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-STRESSNG.json [hunk 4]
```diff
@@ -4,15 +4,18 @@
...
+    },
+    "Parameters": {
+        "CommandLine": "--timeout 60"
```
[the remark this answers is not in the document]
Comment (imadityaa) [reply src/VirtualClient/VirtualClient.Main/profiles/PERF-STRESSNG.json:11]: yes, tested with it. --cpu-load is for load per cpu, while --cpu/-c is for number of CPUs. By default, we will stress all CPUs to max as would be expectation from Stress-Ng, but user can reduce that as per requirement with control over command line

File: src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs (hunk 1 above)
Comment (imadityaa) [reply src/VirtualClient/VirtualClient.Actions/StressNg/StressNgExecutor.cs]: There the parameter is --cpu-load. --cpu/-c is for cpu count

**Review by deep1712 — APPROVED** [review]

**Pull request merged**